pkg/AnewLibrary/Fields/Atoms/TextField.js

```diff
@@ -4,14 +4,21 @@ export const atom = (log, resolve) => ({
  * Copyright 2023 Atom54 LLC
  * SPDX-License-Identifier: BSD-3-Clause
  */
-update({value, form}, state, {service, isDirty}) {
+update({value, form, options}, state, {service, isDirty}) {
   service('FormService', 'RegisterField', {form});
   if (isDirty('value')) {
     state.value = value;
     return {value};
   }
+  if (options && isDirty('options')) {
```

Comment:
I don't see a reason to check `isDirty` here. Also, you should put this before the previous clause, because the previous clause may return early.

Reply:
good catch, moved the if {} above. 
I kept `isDirty` for now, because there is no need to reset `state.options` if they haven't changed. That said, if `isDirty` is more expensive than `JSON.parse` i should remove it.